linter/src/rules/prefer_robust_stmts.rs

```diff
@@ -20,9 +20,13 @@ enum Constraint {
 /// more robust by using guards like `IF NOT EXISTS`. So if the migration fails
 /// halfway through, it can be rerun without human intervention.
 #[must_use]
-pub fn prefer_robust_stmts(tree: &[RawStmt], _pg_version: Option<Version>) -> Vec<RuleViolation> {
+pub fn prefer_robust_stmts(
+    tree: &[RawStmt],
+    _pg_version: Option<Version>,
+    assume_in_transaction: bool,
+) -> Vec<RuleViolation> {
     let mut errs = vec![];
-    let mut inside_transaction = false;
+    let mut inside_transaction = assume_in_transaction;
```

Comment:
ditto here with regards to nested transactions -- which aren't valid in postgres right?

maybe we error or something?

Reply:
I reviewed this again carefully and feel that the behavior is correct in the presence/absence of --assume-in-transaction and also if there are repeated BEGIN/COMMIT statements.